### PR TITLE
Presets: Replace `config` with `previewAnnotations`, remove `previewEntries`

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: We realised that the handling of the `previewEntries` field in the vite build wasn't quite right (it was treating them identically to `config` entries, which have special behaviour on export).

Looking in the code base, we no longer use `previewEntries` (did we ever?), so decided to essentially pseudo-deprecate them by not supporting them in the Vite builder.

## How to test

Check a vite sandbox.

### CODE REVIEW

File: scripts/sandbox.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import {
 import prompts from 'prompts';
 import type { AbortController } from 'node-abort-controller';
 import command from 'execa';
+import dedent from 'ts-dedent';
```
Comment (IanVS) [inline scripts/sandbox.ts:16]: We should add https://www.npmjs.com/package/@ianvs/prettier-plugin-sort-imports so we don't need to worry about manually organizing imports.  :-D

**Review by IanVS — APPROVED** [review]
Looks good to me, thanks!

**Review by shilman — COMMENTED** [review]
@tmeasday @IanVS 

What's the difference between `previewEntries` and `config`? Is it that `config` exports are treated as annotations (e.g. `parameters` etc.)?

I think @ndelangen and I may have introduced `previewEntries` as an intended replacement for `config`. Since `config` doesn't actually mean anything, and `previewEntries` does. Since we're in a breaking release, I'd propose that we take the opportunity to sort this out properly, even if it means introducing a new preset (`previewAnnotations`?)

WDYT?

**Comment by tmeasday** [comment]
> What's the difference between previewEntries and config? Is it that config exports are treated as annotations (e.g. parameters etc.)?

Yes that's the difference.

> I think @ndelangen and I may have introduced previewEntries as an intended replacement for config. 

I don't believe so, but @ndelangen could say for sure. 

This is where they are pulled in by the webpack builder: 

https://github.com/storybookjs/storybook/blob/1335697e9bb12706360abdcbacfa2eccfd7e3b72/code/lib/builder-webpack5/src/presets/preview-preset.ts#L8

Actually I had it wrong, `previewEntries` *is* used in on place by our code: https://github.com/storybookjs/storybook/blob/1335697e9bb12706360abdcbacfa2eccfd7e3b72/code/lib/core-server/src/presets/common-preset.ts#L45-L49

I would suggest we replace it with a `config` (or, as you say a `previewAnnotations`) and not bother maintaining two subtly difference concepts (just make sure we don't export if we don't want to provide annotations). WDYT @ndelangen ?

**Comment by tmeasday** [comment]
- Replace above `previewEntries` with a `config`
- Rename `config` to `previewAnnotations`
- Deprecate `config` + warn.
- Add migration notes

**Comment by tmeasday** [comment]
Note `previewAnnotations` *was* added in https://github.com/storybookjs/storybook/pull/17755 @shilman, however the builder code all still uses `config` to get it.

**Comment by tmeasday** [comment]
Add to migration notes

**Review by shilman — APPROVED** [review]
Looking good!

**Pull request merged**